h/session.py

```diff
@@ -75,22 +75,12 @@ def _current_groups(request, authority):
     """
 
     user = request.user
-    authority_groups = (request.find_service(name='authority_group')
-                        .public_groups(authority=authority))
-
-    groups = authority_groups + _user_groups(user, request)
+    svc = request.find_service(name='list_groups')
+    groups = svc.all_groups(user=user, authority=authority)
 
     return [_group_model(request.route_url, group) for group in groups]
```

Comment:
Note that `session` retains its unique formatting of groups (cf. the `GroupJSONPresenter` used by the `api_groups` view). This means that the API response groups at `GET /profile/groups` look different than those at `GET /groups`. This is on purpose for now as not to break things more than necessary at one time.